### PR TITLE
Updated samples to use latest boot

### LINKED ISSUE #289: Update task samples to boot 1.5.2
Also update tests to use updated stream paradigms as well.

### PR DESCRIPTION
**Pull request by cppwfs**
Update integration tests to use latest stream paradigm
Update tests to be in compliance with 1.5.2.

resolves #289

### CODE REVIEW

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java [hunk 1]
```diff
@@ -240,7 +241,7 @@ public void receive(Object exceptionMessage) {
 	public static class ItemWriteEventsListenerBinding {
 
 		@StreamListener(Sink.INPUT)
-		public void receive(Object itemWrite) {
+		public void receive(String itemWrite) {
 			assertTrue("Message should start with '3 items'", itemWrite.toString().startsWith("3 items "));
```
Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java]: @mbogoevici Is the requirement to make these changes expected?  No objections, just want to be sure that this isn't an unexpected side effect from something.

Comment (mbogoevici) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java]: @mminella Does not seem like a side effect - basically the argument was always a String, so now the method is just treating it as such (StreamListener helps with the coercion in certain cases). Doesn't seem to make a difference when the argument is not actually being used.

Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java]: @mbogoevici What I'm specifically asking is that, before `Object` worked.  Now we're being required to specify the type (the test fails without this change).

Comment (mbogoevici) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java]: That's a little odd then. Maybe we can take a look.

Comment (mbogoevici) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/listener/BatchExecutionEventTests.java]: @mminella The test failures were called by a regression - see https://github.com/spring-cloud/spring-cloud-stream/issues/844. 

On the other hand, I would recommend keeping these changes, as they are closer to what a user would be expected to do anyway (i.e. not use Object and downcast, but use the actual expected type as argument).

**Comment by mminella** [comment]
This PR isn't building for me:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/mminella/Documents/IntelliJWorkspace/spring-cloud-task/spring-cloud-task-stream/src/test/java/org/springframework/cloud/task/launcher/configuration/TaskConfiguration.java:[40,23] org.springframework.cloud.task.launcher.configuration.TaskConfiguration.TestTaskLauncher is not abstract and does not override abstract method environmentInfo() in org.springframework.cloud.deployer.spi.task.TaskLauncher
[INFO] 1 error
```

**Comment by mminella** [comment]
Merged as 4e68adf5facdc92faff5384af65846faa298848d

**Pull request closed without merging**